mscore/pagesettings.cpp

```diff
@@ -109,158 +111,211 @@ void PageSettings::setScore(Score* s)
+      updateWidthHeight(odd->fullRect());
       }
 
 //---------------------------------------------------------
-//   inchClicked
+//   updateWidthHeight - updates Width and Height widgets
 //---------------------------------------------------------
 
-void PageSettings::inchClicked()
+void PageSettings::updateWidthHeight(const QRectF& rect)
       {
-      mmUnit = false;
-      updateValues();
+      pageWidth ->blockSignals(true);
+      pageHeight->blockSignals(true);
+      pageWidth ->setValue(rect.width());
+      pageHeight->setValue(rect.height());
+      pageWidth ->blockSignals(false);
+      pageHeight->blockSignals(false);
       }
 
 //---------------------------------------------------------
-//   mmClicked
+//   typeChanged - populates sizesList based on typesList
 //---------------------------------------------------------
 
-void PageSettings::mmClicked()
+void PageSettings::typeChanged(int idx)
       {
-      mmUnit = true;
-      updateValues();
+      switch (idx) {
+      case PAPER_TYPE_METRIC:
+            _sizes = &MScore::sizesMetric;
+            break;
+      case PAPER_TYPE_IMPERIAL:
+            _sizes = &MScore::sizesImperial;
+            break;
+      case PAPER_TYPE_OTHER:
+            _sizes = &MScore::sizesOther;
+            break;
+      }
+      sizesList->clear(); ///!!!Custom is back at the top of the list
+      sizesList->addItem(QPageSize::name(QPageSize::Custom), int(QPageSize::Custom));
+      for (std::set<int>::iterator i = _sizes->begin(); i != _sizes->end(); ++i)
+            sizesList->addItem(QPageSize::name(QPageSize::PageSizeId(*i)), *i);
       }
 
 //---------------------------------------------------------
-//   orientationClicked
-//    swap width/height
+//   setPageSize - a helper for page size, width, height changes
 //---------------------------------------------------------
 
-void PageSettings::orientationClicked()
+void PageSettings::setPageSize(QPageSize::PageSizeId psid)
       {
-      qreal w = preview->score()->styleD(Sid::pageWidth);
-      qreal h = preview->score()->styleD(Sid::pageHeight);
+      MStyle&         style = preview->score()->style();
+      QPageSize::Unit unit  = QPageSize::Unit(style.pageOdd()->units());
+      QPageSize*      qps;
 
-      preview->score()->style().set(Sid::pageWidth, h);
-      preview->score()->style().set(Sid::pageHeight, w);
+      bool isPreset = (psid != QPageSize::Custom);
+      if (isPreset)
+            qps = new QPageSize(psid);
+      else
+            qps = new QPageSize(QSizeF(pageWidth->value(), pageHeight->value()),
+                  unit,
+                  QPageSize::name(psid),
+                  QPageSize::ExactMatch);
+
+      style.setPageSize(qps);
+      style.pageOdd( )->setPageSize(*qps);
+      style.pageEven()->setPageSize(*qps);
+
+      if (isPreset) {
+            QRectF rect = qps->rect(unit);
+            updateWidthHeight(rect);
+            updatePreview();
+      }
```

Comment:
This brace should be indented one more level.

Reply:
check